src/js/integrations/rewriteframes.ts

```diff
@@ -1,28 +1,42 @@
 import { RewriteFrames } from '@sentry/integrations';
 import type { StackFrame } from '@sentry/types';
+import { Platform } from 'react-native';
+
+import { isExpo } from '../utils/environment';
 
 /**
  * Creates React Native default rewrite frames integration
  * which appends app:// to the beginning of the filename
- * and removes file://, 'address at' prefixes and CodePush postfix.
+ * and removes file://, 'address at' prefixes, CodePush postfix,
+ * and Expo bundle postfix.
  */
 export function createReactNativeRewriteFrames(): RewriteFrames {
   return new RewriteFrames({
     iteratee: (frame: StackFrame) => {
-      if (frame.filename) {
-        frame.filename = frame.filename
-          .replace(/^file:\/\//, '')
-          .replace(/^address at /, '')
-          .replace(/^.*\/[^.]+(\.app|CodePush|.*(?=\/))/, '');
+      if (!frame.filename) {
+        return frame;
+      }
+      delete frame.abs_path;
+
+      frame.filename = frame.filename
+        .replace(/^file:\/\//, '')
+        .replace(/^address at /, '')
+        .replace(/^.*\/[^.]+(\.app|CodePush|.*(?=\/))/, '');
 
-        if (frame.filename !== '[native code]' && frame.filename !== 'native') {
-          const appPrefix = 'app://';
-          // We always want to have a triple slash
-          frame.filename =
-            frame.filename.indexOf('/') === 0 ? `${appPrefix}${frame.filename}` : `${appPrefix}/${frame.filename}`;
-        }
-        delete frame.abs_path;
+      if (frame.filename === '[native code]' || frame.filename === 'native') {
+        return frame;
       }
+
+      // Expo adds hash to the end of bundle names
+      if (isExpo()) {
+        frame.filename = Platform.OS === 'android' ? 'app:///index.android.bundle' : 'app:///main.jsbundle';
```

Comment:
M: Since it's technically possible to run on Web/Desktop as well, better to check if `Platform.OS` is from the Apple ecosyste, and fallback to something else.